source/agora/crypto/Schnorr.d

```diff
@@ -121,17 +121,55 @@ public Point extractNonce (Signature sig) @safe @nogc nothrow pure
 
 *******************************************************************************/
 
-public struct Sig
+public struct Signature
 {
+    import geod24.bitblob;
+
+    import std.format;
+
     /// Commitment
     public Point R;
     /// Proof
     public Scalar s;
 
-    static assert(Signature.sizeof == typeof(this).sizeof);
+    static assert(BitBlob!64.sizeof == typeof(this).sizeof);
+
+    /// construct from Point and Scalar
+    public this (in Point R, Scalar s) pure nothrow @nogc @safe
+    {
+        this.R = R;
+        this.s = s;
+    }
+
+    /// construct from its string representation
+    public this (in char[] str)
+    {
+        auto offset = str.length == 130 ? 2 : 0;
+        assert(str[offset .. $].length == 128, format!"Signature hex string should be 128 characters (or 130 with 0x) not %s"(str[offset .. $].length));
+        this.R = Point(str[offset .. offset + 2 * Point.sizeof]);
+        this.s = Scalar(str[offset + 2 * Point.sizeof .. $]);
+    }
```

Comment:
Is this used ? Same question for the other ctor.

Reply:
I removed this function although do have a static `fromString` to create Signatures from hex strings.